src/librustc/hir/ptr.rs

```diff
@@ -24,8 +24,8 @@ pub fn P<T: 'static>(value: T) -> P<T> {
 
 impl<T: 'static> P<T> {
     // HACK(eddyb) used by HIR lowering in a few places still.
-    // NOTE: do not make this more public than `pub(super)`.
-    pub(super) fn into_inner(self) -> T {
+    // NOTE: do not use outside HIR lowering!
+    pub fn into_inner_only_use_in_lowering(self) -> T {
         *self.ptr
     }
 }
```

Comment:
These changes are going away very soon in the 4th arena-ification PR so it shouldn't be an issue.